widgetastic_manageiq/__init__.py

```diff
@@ -4051,3 +4051,149 @@ def data_for_timestamp(self, timestamp):
+        """Method to read elements that are currently selected."""
+        return self.list_selected_items
+
+    def fill(self, values, partial_match=False):
+        """Click/Select multiple the items in the multiselectbox
+
+        Args:
+            items: 'list' of items of type 'str' to be matched
+            partial_match: if you want to use partial text matching. Default:False
+
+        Returns:
+            True if all values were found and clicked.
+        """
+        # TODO self.browser.plugin.ensure_page_safe() should be here
+        wait_for(lambda: "loading" not in self.browser.classes(self), timeout=5)
+        if values == self.list_selected_items or values == []:
+            return False
+        if values:
+            for item in values:
+                self.select_by_text(item, partial_match=partial_match)
+        return True
+
+    def select_by_text(self, text, partial_match=False):
+        """Select/click option using text
+
+        Args:
+            text(str): text to be matched
+            partial_match: if you want to use partial text matching. Default:False
+        """
+        try:
+            if text and not partial_match:
+                self.logger.info("Not using partial text matching to find %s", text)
+                self._get_all_options()[text].click()
+
+            elif text and partial_match:
+                self.logger.info("Using partial text matching to find %s", text)
+                all_options = self._get_all_options()
+                for key in all_options:
+                    if text in key:
+                        all_options[key].click()
+                        break
+        except KeyError:
+            raise ItemNotFound("{} was not found in {}".format(text, self.list_items))
+
+
+class InfraMappingTreeView(Widget):
+    """Widget to capture/interact with the InfraMappingTreeView in Infra Mapping Wizard for v2v.
+
+    Args:
+        tree_class(str): Class name of the div representing this TreeView
+    """
+    ROOT = ParametrizedLocator('.//div[contains(@class,{@tree_class|quote})]')
+    ROOT_ITEMS = './/ul/li[./span[contains(@class,"expand-icon")] and position() > 1]'
+    ITEMS_LOCATOR = './/ul/li[./span[contains(@class, "glyphicon")] and position() > 1]'
+    SELECTED_NODE = './/li[contains(@class, "node-selected")]'
+
+    def __init__(self, parent, tree_class=None, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.tree_class = tree_class
+
+    @property
+    def mapping_targets(self):
+        """Returns list of Mapping targets or in other words, leaf nodes."""
+        return [self.browser.text(el) for el in self.browser.elements(self.ROOT_ITEMS)]
+
+    @property
+    def mapping_sources(self):
+        """Returns list of Mapping sources or in other words, parent nodes to leaf nodes."""
+        return [self.browser.text(el) for el in self.browser.elements(self.ITEMS_LOCATOR)]
+
+    @property
+    def selected_item(self):
```

Comment:
@quarckster this is returning NoSuchElement if nothing is selected. Do you want to handle that or let it throw?